corehq/apps/locations/tests/test_bulk_management.py

```diff
@@ -0,0 +1,652 @@
+                ('D', 'd', 'a', False, False, False, 'b', 'd', 0),
+                ('E', 'e', 'b', False, False, False, 'b', '', 0),
+            ],
+            []
+        )
+        errors = validator.errors
+        self.assertEqual(len(errors), 2)
+
+    def test_duplicate_location(self):
+        validator = get_validator(FLAT_LOCATION_TYPES, DUPLICATE_SITE_CODES)
+        errors = validator.errors
+        self.assertEqual(len(errors), 2)
+        self.assertEqual(len(validator._check_unique_location_codes()), 1)
+        self.assertEqual(len(validator._check_unique_location_ids()), 1)
+        self.assertIn("cambridge", errors[0])
+
+    def test_same_name_same_parent(self):
+        validator = get_validator(FLAT_LOCATION_TYPES, SAME_NAME_SAME_PARENT)
+        errors = validator.errors
+        self.assertEqual(len(errors), 1)
+        self.assertEqual(len(validator._check_location_names()), 1)
+        self.assertIn("middlesex", errors[0])
+
+    def test_missing_types(self):
+        # all types in the domain should be listed in given excel
+        old_types = FLAT_LOCATION_TYPES + [('Galaxy', 'galaxy', '', False, False, False, '', '', 0)]
+
+        old_collection = make_collection(old_types, BASIC_LOCATION_TREE)
+        validator = get_validator(FLAT_LOCATION_TYPES, BASIC_LOCATION_TREE, old_collection)
+
+        missing_type_errors = validator._check_unlisted_type_codes()
+        self.assertEqual(len(missing_type_errors), 1)
+        self.assertEqual(len(validator.errors), 1)
+        self.assertIn('galaxy', missing_type_errors[0])
+
+    def test_missing_location_ids(self):
+        # all locations in the domain should be listed in given excel
+        old_locations = (
+            BASIC_LOCATION_TREE +
+            [('extra_state', 'ex_code', 'state', '', 'ex_id', False, '', '', '', 0)]
+        )
+        old_collection = make_collection(FLAT_LOCATION_TYPES, old_locations)
+        validator = get_validator(FLAT_LOCATION_TYPES, BASIC_LOCATION_TREE, old_collection)
+        missing_locations = validator._check_unlisted_location_ids()
+        self.assertEqual(len(missing_locations), 1)
+        self.assertEqual(len(validator.errors), 1)
+        self.assertIn('extra_state', missing_locations[0])
+
+    def test_unknown_location_ids(self):
+        # all locations in the domain should be listed in given excel
+
+        old_collection = make_collection(FLAT_LOCATION_TYPES, BASIC_LOCATION_TREE)
+        new_locations = (
+            BASIC_LOCATION_TREE +
+            [('extra_state', 'ex_code', 'state', '', 'ex_id', False, '', '', '', 0)]
+        )
+        validator = get_validator(FLAT_LOCATION_TYPES, new_locations, old_collection)
+        unknown_locations = validator._check_unknown_location_ids()
+        self.assertEqual(len(unknown_locations), 1)
+        self.assertEqual(len(validator.errors), 1)
+        self.assertIn('ex_id', unknown_locations[0])
+
+
+class TestBulkManagement(TestCase):
```

Comment:
@esoergel What do you think of overall test coverage, do you think anything more is needed?